src/evergreen/api.py

```diff
@@ -730,15 +709,21 @@ def get_api(cls, auth=None, use_config_file=False, config_file=None,
         :return: EvergreenApi instance.
         """
         kwargs = {'auth': auth, 'timeout': timeout}
-        if not auth and use_config_file:
-            config = read_evergreen_config()
-            auth = get_auth_from_config(config)
-            if auth:
-                kwargs['auth'] = auth
-
-            # If there is a value for api_server_host, then use it.
-            if 'evergreen' in config and config['evergreen'].get('api_server_host', None):
-                kwargs['api_server'] = config['evergreen']['api_server_host']
+        if not auth:
```

Comment:
We might want to consider pulling this logic into a separate function. That would make it easier to test and remove the need to mock as much.